### PR TITLE
[backend/frontend] Fix editing challenge

### PR DESCRIPTION
**Pull request by RomuDeuxfois**
* Closes https://github.com/OpenBAS-Platform/openbas/issues/1758

### CODE REVIEW

**Comment by isselparra** [comment]
TEST OK
- Creation of Challenge -> OK, No regression
- Update of a Challenge -> OK, fix works

I have a question regarding the values of Score and Max number of attempts, is it normal that I'm able to set negative numbers?
I updated my Challenge with a -0.43 Score and a -42 as Max number of attempts
Screenshot from 2024-11-19 16-14-38

File: openbas-api/src/main/java/io/openbas/rest/challenge/form/ChallengeInput.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotEmpty;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class ChallengeInput {
+
```
[the remark this answers is not in the document]
Comment (RomuDeuxfois) [reply openbas-api/src/main/java/io/openbas/rest/challenge/form/ChallengeInput.java:15]: Good point

File: openbas-api/src/test/java/io/openbas/rest/ChallengeApiTest.java [hunk 2]
```diff
@@ -10,47 +13,41 @@
...
-import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.http.MediaType;
 import org.springframework.test.web.servlet.MockMvc;
 
-@SpringBootTest
-@AutoConfigureMockMvc
 @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 @TestInstance(PER_CLASS)
-class ChallengeApiTest {
+class ChallengeApiTest extends IntegrationTest {
```
[the remark this answers is not in the document]
Comment (RomuDeuxfois) [reply openbas-api/src/test/java/io/openbas/rest/ChallengeApiTest.java:34]: for next time

**Comment by isselparra** [comment]
TEST OK 

- Creation of Challenge -> OK, No regression
- Update of a Challenge -> OK, fix works

Just a last question, to be sure, a Score can only be an integer ? In the DB it's indicated as `double precision`

**Review by isselparra — APPROVED** [review]

**Pull request merged**